analysis/cross_partition_combiners.py

```diff
@@ -254,61 +257,85 @@ def _merge_utility_reports(report1: metrics.UtilityReport,
         _merge_metric_utility(utility1, utility2)
 
 
-def _average_utility_report(report: metrics.UtilityReport,
-                            public_partitions: bool,
-                            sums_actual: Tuple) -> None:
+def _average_utility_report(report: metrics.UtilityReport, sums_actual: Tuple,
+                            total_weight: float) -> None:
     """Averages fields of the 'report' across partitions."""
     if not report.metric_errors:
         return
-    partitions = report.partitions_info
-    if public_partitions:
-        num_output_partitions = partitions.num_dataset_partitions + partitions.num_empty_partitions
-    else:
-        num_output_partitions = partitions.kept_partitions.mean
 
     for sum_actual, metric_error in zip(sums_actual, report.metric_errors):
         _multiply_float_dataclasses_field(
             metric_error,
-            1.0 / num_output_partitions,
+            1.0 / total_weight,
             fields_to_ignore=["noise_std", "ratio_data_dropped"])
         scaling_factor = 1 if sum_actual == 0 else 1.0 / sum_actual
         _multiply_float_dataclasses_field(metric_error.ratio_data_dropped,
                                           scaling_factor)
 
 
+def partition_size_weight_fn(
+        per_partition_metrics: metrics.PerPartitionMetrics) -> float:
+    # Only one metric is calculated as of now.
+    return per_partition_metrics.metric_errors[0].sum
+
+
+def equal_weight_fn(
+        per_partition_metrics: metrics.PerPartitionMetrics) -> float:
+    """
```

Comment:
Nit: docstring format

"""1 line description

[longer description]
""""

Here it looks like comment explain technical details, so it's better to have it as a comment to implementation. 
Please add docstring (1 line is enough) and make this comment to normal (not-docstring) comment